go/vt/vtgate/resilient_srv_topo_server.go

```diff
@@ -280,33 +302,64 @@ func (server *ResilientSrvTopoServer) GetSrvKeyspace(ctx context.Context, cell,
 	entry.mutex.Lock()
 	defer entry.mutex.Unlock()
 
-	// If the entry is fresh enough, return it
-	if time.Now().Sub(entry.insertionTime) < server.cacheTTL {
+	// If the entry exists, return it
+	if !entry.insertionTime.IsZero() {
 		return entry.value, entry.lastError
 	}
 
-	// not in cache or too old, get the real value
+	// not in cache, get the real value
 	newCtx, cancel := context.WithTimeout(context.Background(), *srvTopoTimeout)
 	defer cancel()
 
-	result, err := server.topoServer.GetSrvKeyspace(newCtx, cell, keyspace)
+	// start watching
+	notifications, _, err := server.topoServer.WatchSrvKeyspace(newCtx, cell, keyspace)
```

Comment:
By the way, looking at it now, maybe the WatchSrvKeyspace API is wrong: instead of the stopWatching channel that is returned, WatchSrvKeyspace should just close the notifications channel and return if the context is cancelled. That way the caller can just cancel the context to stop the watch, and that triggers the notification channel to be closed. So the go routine that waist on notifications can just range on the channel.